src/modules/lua-support/mtev/Proc.lua

```diff
@@ -0,0 +1,287 @@
+    error("Could not start process")
+  end
+  self.proc = proc
+  in_e:close()
+  out_e:close()
+  self.key_ready = "proc-ready-" .. mtev.uuid()
+  mtev.coroutine_spawn(function()
+      local err_e = err_e:own()
+      local ready = false
+      while true do
+        local line = err_e:read("\n")
+        if line == nil then
+          if not ready then mtev.notify(self.key_ready, false) end
+          break
+        end
+        for _, write in ipairs(self.log_writers) do
+          write(line)
+        end
+        for key, watcher in pairs(self.log_watchers) do
+          if watcher.matches(line) then
+            mtev.notify(key, line)
+            if watcher.limit then
+              watcher.limit = watcher.limit - 1
+              if watcher.limit <= 0 then
+                self.log_watchers[key] = nil
+              end
+            end
+          end
+        end
+        if not ready and line:find(self.boot_match) then
+          mtev.notify(self.key_ready, true)
+          ready = true
+        end
+      end
+      for _, hook in ipairs(self.hook_term) do
+        hook()
+      end
+      return
+  end)
+  return self
+end
+
+--/*!
+--\hlua status = Proc:ready()
+--\brief wait for the process to become ready
+--\return status true/false depending on weather the process became ready
+--Kills processes that did not become ready in time
+--*/
+function Proc:ready()
+  local key, ok = mtev.waitfor(self.key_ready, self.boot_timeout)
+  if not ok then -- cleanup process that could not start
+    assert(self:kill(10), "Failed to failing kill child process")
+    self.proc = nil
+    return false
+  end
+  return true
+end
+
+--/*!
+--\hlua ok, status, errno = Proc:kill(timeout)
+--\brief Kill process by sending SIGTERM, then SIGKILL
+--\param timeout for the signals
+--\return ok true if process was terminated, status, errno as returned by mtev.proc:wait()
+--*/
+function Proc:kill(timeout)
+  timeout = timeout or 2
+  if self.proc == nil or self.proc:pid() == -1 then return end
+  local ok, errno = self.proc:kill()
+  if not ok then error("Faild to deliver kill signal rv=" .. tostring(errno)) end
+  local term, status, errno = self:wait(timeout)
+  if term then return true, status, errno end
+  local ok, errno = self.proc:kill(9)
+  if not ok then error("Failed to deliver kill signal rv=" .. tostring(errno)) end
+  return self:wait(timeout)
+end
+
+--/*!
+--\hlua pid = mtev.Proc:pid()
+--*/
+function Proc:pid()
+  if self.proc == nil then return -1 end
+  return self.proc:pid()
+end
+
+--/*!
+--\hlua term, status, errno = mtev.Proc:wait(timeout)
+--\brief wait for a process to terminate
+--\return trem is true if the process terminated and status, errno as returned by mtev.process:wait()
+--*/
+function Proc:wait(timeout)
+  if self.proc == nil then return nil end
+  local status, errno = self.proc:wait(timeout)
+  if status then
+    if mtev.WIFEXITED(status) or mtev.WIFSIGNALED(status) then
+      return true, status
```

Comment:
Probably want to return `mtev.WEXITSTATUS(status)` here and below

Reply:
Good point. Does that also work if WIFSIGNALED was true?

Reply:
nope, from docs: 
```       
WEXITSTATUS(wstatus)
              returns the exit status of the  child.   This  consists  of  the
              least  significant  8 bits of the status argument that the child
              specified in a call to exit(3) or _exit(2) or  as  the  argument
              for a return statement in main().  This macro should be employed
              only if WIFEXITED returned true.
```

Reply:
Yeah, makes sense.

I found returning the status struct to be bad usability, so I was looking for something better.
Usually I am interested in: "Is the process dead or not?"
So, this is what I (tried?) to return in first return value. You also get the full status struct and error info in case you want to start digging.

Not the greatest API I have seen. Not sure how to improve it though.